Add a section about {dotnet_root} requirements (#32208)
dotnet/docs: docs/core/distribution-packaging.md

When installed, .NET consists of several components that are laid out as follows in the file system:

```

> {dotnet_root}                    (0)

Reviewer: I think the asterias should be maintained as the `dotnet_root` is used by multiple packages.

```suggestion
{dotnet_root}                    (0)               (*)
```

        └── dotnet               (10)
```

> - (0) **{dotnet_root}** is a shared root for all .NET major and minor versions. For example, if multiple runtimes are installed, they would share the **{dotnet_root}** folder, `{dotnet_root}/shared/Microsoft.NETCore.App/6.0.11` and `{dotnet_root}/shared/Microsoft.NETCore.App/7.0.0`. It is advised that the name of `{dotnet_root}` folder be version agnostic, i.e. simply `dotnet`.

Reviewer 2: ```suggestion
- (0) **{dotnet_root}** is a shared root for all .NET major and minor versions. If multiple runtimes are installed, they share the **{dotnet_root}** folder, for example, `{dotnet_root}/shared/Microsoft.NETCore.App/6.0.11` and `{dotnet_root}/shared/Microsoft.NETCore.App/7.0.0`. The name of the `{dotnet_root}` folder should be version agnostic, that is, simply `dotnet`.
```

The folders marked with `(*)` are used by multiple packages. Some package formats (for example, `rpm`) require special handling of such folders. The package maintainer must take care of this.

> - (18) **{dotnet_root}** is a shared root for all .NET major and minor versions. As a consequence, you will see all diffrent runtime versions, for all .NET major and minor versions, under a single folder on user's machine - `{dotnet_root}/shared/Microsoft.NETCore.App`. The same is true for all other folders that contain version sub-folders, i.e. `{dotnet_root}/shared/sdk`. It is strongly advised that the name of `{dotnet_root}` folder be version agnostic, i.e. simply `dotnet`.

Reviewer: I'm not finding the `(18)` reference in the above layout.  The numbers here are cross references to the above layout.  I think 18 needs to be added.

Author: I see, let me check how to update the table. I was thinking of adding this new section to be the first one, but feared the big diff due to renumbering. I'll see if {0} is also possible.

> - (18) **{dotnet_root}** is a shared root for all .NET major and minor versions. As a consequence, you will see all diffrent runtime versions, for all .NET major and minor versions, under a single folder on user's machine - `{dotnet_root}/shared/Microsoft.NETCore.App`. The same is true for all other folders that contain version sub-folders, i.e. `{dotnet_root}/shared/sdk`. It is strongly advised that the name of `{dotnet_root}` folder be version agnostic, i.e. simply `dotnet`.

Reviewer: ```suggestion
- (18) **{dotnet_root}** is a shared root for all .NET major and minor versions. For example, if multiple runtimes are installed, they would share the **{dotnet_root}** folder, `{dotnet_root}/shared/Microsoft.NETCore.App/6.0.11` and `{dotnet_root}/shared/Microsoft.NETCore.App/7.0.0`. It is advised that the name of `{dotnet_root}` folder be version agnostic, i.e. simply `dotnet`.
```

Author: I've incorporated this in the second commit. I've renamed {18} to {0} and updated the layout accordingly.
